platform/javascript/detect.py

```diff
@@ -104,44 +124,31 @@ def configure(env):
     env['LIBPREFIXES'] = ['$LIBPREFIX']
     env['LIBSUFFIXES'] = ['$LIBSUFFIX']
 
-    ## Compile flags
-
     env.Prepend(CPPPATH=['#platform/javascript'])
     env.Append(CPPDEFINES=['JAVASCRIPT_ENABLED', 'UNIX_ENABLED'])
 
-    # No multi-threading (SharedArrayBuffer) available yet,
-    # once feasible also consider memory buffer size issues.
-    env.Append(CPPDEFINES=['NO_THREADS'])
-
-    # Disable exceptions and rtti on non-tools (template) builds
-    if not env['tools']:
-        # These flags help keep the file size down.
-        env.Append(CCFLAGS=['-fno-exceptions', '-fno-rtti'])
-        # Don't use dynamic_cast, necessary with no-rtti.
-        env.Append(CPPDEFINES=['NO_SAFE_CAST'])
-
     if env['javascript_eval']:
         env.Append(CPPDEFINES=['JAVASCRIPT_EVAL_ENABLED'])
 
-    ## Link flags
+    # Thread support (via SharedArrayBuffer).
+    if env['threads_enabled']:
+        env.Append(CPPDEFINES=['PTHREAD_NO_RENAME'])
+        env.Append(CPPFLAGS=['-s', 'USE_PTHREADS=1']);
```

Comment:
Is this only relevant for C++ code (`CPPFLAGS` vs `CCFLAGS`)?

Reply:
Only relevant for c/c++ code, it enable support for `<mutex>` `<thread>` and `<pthread.h>`.
It should become CCFLAGS then, right?

Reply:
Wait, my bad. I was again confused by the weird naming of the SCons flags variables. `CPPFLAGS` is for C (and C++) preprocessor options, it's `CXXFLAGS` which is for C++ compiler flags only (`CFLAGS` is C compiler flags and `CCFLAGS` is both C and C++ compiler flags).

So if this is meant to be passed to the preprocessor, it's correct as is.